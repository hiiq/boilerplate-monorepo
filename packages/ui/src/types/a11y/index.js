import { Role } from './role';

export const A11y = {
  Role,
};
