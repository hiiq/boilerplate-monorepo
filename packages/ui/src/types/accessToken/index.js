// This variable is the only place that the access token is stored
let token = null;

const clear = () => {
  token = null;
};
const read = () => {
  return token;
};
const set = newToken => {
  token = newToken;
};

export const AccessToken = {
  clear,
  read,
  set,
};
