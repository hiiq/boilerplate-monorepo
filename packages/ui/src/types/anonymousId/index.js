import { LocalStorage } from '@boilerplate-monorepo/ui-common';
import cuid from 'cuid';

const makeId = () => `anonymous-${cuid()}`;
const clear = () => localStorage.removeItem(LocalStorage.ANONYMOUS_ID);

const set = () => {
  const newId = makeId();

  localStorage.setItem(LocalStorage.ANONYMOUS_ID, newId);

  return newId;
};

const read = () => localStorage.getItem(LocalStorage.ANONYMOUS_ID) || set();

export const AnonymousId = {
  clear,
  read,
  set,
};
