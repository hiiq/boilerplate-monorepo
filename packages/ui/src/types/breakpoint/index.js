export const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};
