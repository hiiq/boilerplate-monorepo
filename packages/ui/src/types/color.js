export const Color = {
  black: '#212121',
  border: '#DCDCDC',
  error: '#DB0000',
  focusRing: '#7AACFE',
  muted: '#8D8D8D',
  primary: '#6d4baa',
  success: '#0C9C0C',
  warn: '#DFDF00',
  white: '#FFFFFF',
};
