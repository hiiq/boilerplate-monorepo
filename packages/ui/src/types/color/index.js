import { Enumeration } from './typedef';

export const Color = {
  ...Enumeration,
};
