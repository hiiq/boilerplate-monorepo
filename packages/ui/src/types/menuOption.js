import { noop } from '@puttingpoker/common';
import { func, shape, string } from 'prop-types';

const propTypes = shape({
  label: string.isRequired,
  onClick: func.isRequired,
  text: string.isRequired,
});

const example = () => ({
  label: 'LABEL',
  onClick: noop,
  text: 'TEXT',
});

export const MenuOption = {
  example,
  propTypes,
};
