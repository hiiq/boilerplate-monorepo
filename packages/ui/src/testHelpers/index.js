import { render } from '@testing-library/react';
import { node } from 'prop-types';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components/macro';
import { Theme } from 'types/theme';

export * from '@testing-library/react';

jest.mock('shared/useTheme');
jest.mock('shared/useTranslate');

const AllTheProviders = ({ children }) => (
  <Router>
    <ThemeProvider theme={Theme.example()}>{children}</ThemeProvider>
  </Router>
);

AllTheProviders.propTypes = {
  children: node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export { customRender as render };
