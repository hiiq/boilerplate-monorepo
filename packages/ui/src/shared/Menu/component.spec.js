import React from 'react';
import { render } from 'testHelpers';
import { MenuOption } from 'types/menuOption';
import { Menu, domTestId } from '.';

/* eslint-disable react/display-name,react/prop-types */
jest.mock('reakit/Menu', () => ({
  Menu: ({ children }) => <x-menu>{children}</x-menu>,
  MenuDisclosure: ({ children }) => (
    <x-menu-disclosure>{children}</x-menu-disclosure>
  ),
  MenuItem: ({ children }) => <x-menu-item>{children}</x-menu-item>,
  useMenuState: () => ({}),
}));
/* eslint-enable react/display-name,react/prop-types */

describe('<Menu/>', () => {
  const defaultProps = {
    children: <x-child />,
    label: 'LABEL',
    menuLabel: 'MENU_LABEL',
    options: [MenuOption.example()],
  };

  it('renders properly', () => {
    const { getByTestId } = render(<Menu {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });
});
