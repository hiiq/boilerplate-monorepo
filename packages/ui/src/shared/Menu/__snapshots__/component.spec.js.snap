// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> renders properly 1`] = `
<x-menu
  data-testid="Menu"
>
  <x-menu-disclosure>
    <x-child />
  </x-menu-disclosure>
  <x-menu>
    <x-menu-item>
      TEXT
    </x-menu-item>
  </x-menu>
</x-menu>
`;
